tools/heir-opt.cpp

```diff
@@ -155,6 +157,72 @@ void polynomialToLLVMPipelineBuilder(OpPassManager &manager) {
   manager.addPass(createSymbolDCEPass());
 }
 
+void tosaToBooleanTfhePipeline(const std::string &yosysFilesPath,
+                               const std::string &abcPath) {
+  PassPipelineRegistration<YosysOptimizerPipelineOptions>(
+      "tosa-to-boolean-tfhe", "Arithmetic modules to boolean tfhe-rs pipeline.",
+      [yosysFilesPath, abcPath](OpPassManager &pm,
+                                const YosysOptimizerPipelineOptions &options) {
+        // Secretize inputs
+        pm.addPass(createSecretize());
+
+        // TOSA to arith pipeline
+        // TOSA to linalg
+        pm.addNestedPass<FuncOp>(createTosaToLinalgNamed());
+        pm.addNestedPass<FuncOp>(createTosaToLinalg());
+        pm.addNestedPass<FuncOp>(createTosaToArith(true, false));
+        pm.addNestedPass<FuncOp>(createTosaToTensor());
+        pm.addPass(bufferization::createEmptyTensorToAllocTensorPass());
+        pm.addNestedPass<FuncOp>(createLinalgDetensorizePass());
+        pm.addPass(createConvertTensorToLinalgPass());
+        pm.addPass(bufferization::createEmptyTensorToAllocTensorPass());
+        // Bufferize
+        pm.addNestedPass<FuncOp>(createLinalgBufferizePass());
+        pm.addNestedPass<FuncOp>(tensor::createTensorBufferizePass());
+        pm.addPass(arith::createArithBufferizePass());
+        pm.addPass(func::createFuncBufferizePass());
+        pm.addNestedPass<FuncOp>(
+            bufferization::createFinalizingBufferizePass());
```

Comment:
Can you try replacing these with one-shot bufferize? Maybe we can extract the sub-pipeline defined in  `polynomialToLLVMPipelineBuilder` for shared use.

Reply:
Factored out bufferizing sub-pipeline, it works!